<?xml version="1.0"?>
<launch>

    <!-- Leds arguments -->
    <arg name="serial_port" default="/dev/ttyACM0" />
    <arg name="baudrate" default="115200" />
    <arg name="velocity_threshold" default= "0.05" />


    <!-- Launch leds driver -->
    <include file="$(find roboticarts_leds)/launch/leds_driver.launch">
        <arg name="serial_port" value="$(arg serial_port)" />
        <arg name="baudrate" value="$(arg baudrate)" />
    </include>


    <!-- Launch teleop leds -->
    <include file="$(find roboticarts_leds)/launch/leds_teleop.launch">
        <arg name="cmd_vel_topic" value="roboticarts_base_control/cmd_vel" />
        <arg name="cmd_led_service" value="roboticarts_teleop/cmd_leds" />
        <arg name="velocity_threshold" value="$(arg velocity_threshold)" />
    </include>


    <!-- Launch leds_mux -->
    <include file="$(find roboticarts_leds)/launch/leds_mux.launch">
        <arg name="leds_mux_config" value="$(find roboticarts_leds)/config/leds_mux.yaml" />
        <arg name="cmd_leds_output" value="leds_driver/set_leds_signal" />
    </include>


</launch>
